ext/liquid_c/vm_assembler.c

```diff
@@ -85,7 +85,7 @@ void vm_assembler_add_write_node(vm_assembler_t *code, VALUE node)
 
 void vm_assembler_add_push_fixnum(vm_assembler_t *code, VALUE num)
 {
-    int x = FIX2INT(num);
+    long x = FIX2LONG(num);
```

Comment:
```c
static inline int
RB_FIX2INT(VALUE x)
{
    long ret;

    if /* constexpr */ (sizeof(int) < sizeof(long)) {
        ret = rb_fix2int(x);
    }
    else {
        ret = RB_FIX2LONG(x);
    }

    return RBIMPL_CAST((int)ret);
}

```

I guess `sizeof(int) < sizeof(long)`?

Reply:
`sizeof(int) < sizeof(long)` is true for our purposes on a 64-bit system, although they are normally both 4 on a 32-bit system.

Reply:
Ah right, it's not that int is small, it's that long is actually long long. So I guess the errors is from

```c
#if SIZEOF_INT < SIZEOF_LONG
static void
check_int(long num)
{
    if ((long)(int)num != num) {
	rb_out_of_int(num);
    }
}
```

Reply:
exactly